frontend/src/components/ForgotPassword.js

```diff
@@ -0,0 +1,126 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import { connect } from 'react-redux'
+import { Redirect, Link } from 'react-router-dom'
+import { Button, Grid, Paper, TextField, Typography } from '@material-ui/core'
+import { withStyles } from '@material-ui/core/styles'
+
+import ErrorMessage from './ErrorMessage'
+import { forgotPassword } from '../actions/auth'
+
+const style = (theme) => ({
+  root: {
+    background: 'linear-gradient(to right, #5390d9, #4ea8de)',
+    display: 'flex',
+    flexDirection: 'column',
+    minWidth: '100%',
+    minHeight: '100vh',
+    alignItems: 'center',
+    justifyContent: 'center'
+  },
+  paper: {
+    alignItems: 'center',
+    minWidth: '460px',
+    padding: theme.spacing.unit * 2
+  },
+  input: {
+    width: '100%'
+  },
+  button: {
+    width: '100%',
+    marginTop: theme.spacing.unit
+  },
+  success: {
+    color: 'green'
+  }
+})
+
+class ForgotPassword extends Component {
+  state = {
+    email: this.props.auth.email
+  };
+
+  validate = () => this.state.email.length > 0;
+
+  handleChange = ({ target: { id, value } }) => this.setState({ [id]: value });
+
+  handleSubmit = (event) => {
+    event.preventDefault()
+    this.props.dispatch(forgotPassword(this.state.email))
+  };
+
+  render () {
+    const { classes } = this.props
+
+    const { sendingResetCode, resetCodeError, resetCodeSent } = this.props.auth
+
+    const validEmail = resetCodeSent ? <Redirect to="/confirm-forgot-password" /> : null
+
+    const errorItem = resetCodeError
+      ? (
+        <Grid item>
+          <ErrorMessage messageId={resetCodeError.id} />
+        </Grid>
+        )
+      : null
+
+    return (
+      <div className={classes.root}>
+        <form onSubmit={this.handleSubmit}>
+          <Paper className={classes.paper}>
+            <Grid
+              className={classes.input}
+              container
+              direction="column"
+              justifyContent="center"
+              alignItems="stretch"
+              spacing={8}
+            >
+              <Grid item>
+                <Typography variant="h3">Forgot Password</Typography>
+              </Grid>
+              <Grid item>
```

Comment:
It would be helpful for the user to add some explanatory text here to let them know how it will work, e.g.,

> When you reset your password, we will send a code to the email address you provide. Once you enter this code, you will be able to select a new password.

Reply:
It is expanded 
<img width="1263" alt="Screenshot 2022-04-07 at 11 07 54" src="https://user-images.githubusercontent.com/74667729/162175840-3a25b5a6-c4ba-468e-a419-7a0b7d25da00.png">

Reply:
We will need to restrict the maximum width